### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by boffbowsh**

### CODE REVIEW

**Comment by veggiemonk** [comment]
:+1:

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM ubuntu:trusty
+MAINTAINER Paul Bowsher <paul.bowsher@gmail.com>
+
+RUN apt-get update
+RUN apt-get install -y sqlite3 libsqlite3-dev git build-essential
+RUN apt-get install -y ruby-dev libssl-dev
+RUN apt-get install -y nodejs
```
Comment (thaJeztah) [inline Dockerfile]: These RUN statements should probably be combined to reduce the number of layers produced and keep image-size down.

Comment (boffbowsh) [inline Dockerfile]: Done

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ADD . /app
+WORKDIR /app
+
+RUN bundle install
+RUN bundle exec rake assets
+
+EXPOSE 1025
+EXPOSE 1080
+
+ENTRYPOINT ["bundle", "exec", "mailcatcher", "--ip", "0.0.0.0", "-f"]
```
Comment (thaJeztah) [inline Dockerfile]: Perhaps:

    ENTRYPOINT ["bundle", "exec", "mailcatcher", "-f"]
    CMD ["--ip", "0.0.0.0"]

**Comment by thaJeztah** [comment]
Made a quick gist: https://gist.github.com/thaJeztah/1d61ab063ee583345670 that

- Puts the installs in alphabetic order (see https://docs.docker.com/articles/dockerfile_best-practices/)
- Cleans up apt cache afterwards

Not yet included the entrypoint idea

**Comment by thaJeztah** [comment]
LOL you're fast. Thanks!

**Comment by thaJeztah** [comment]
Hm, thinking about this; 

   EXPOSE 25 80

and `--http-port 80 --smtp-port=25` Should be possible inside the Dockerfile. You'd be able to connect to the *standard* SMTP ports of the container (e.g. when linked to another container).

If you need to expose those ports on the *host* with the "10XX" ports, you can still do `docker run -p 1080:80 -p 1025:25 ....`

**Comment by thaJeztah** [comment]
But better give it all a test, I'm writing "out of my head"

(the ports should work though, that's how I have configured my mailcatcher Dockerfile)

**Comment by sj26** [comment]
Neat!

`EXPOSE 25 80` won't work because not being bound as root.

**Comment by thaJeztah** [comment]
Well, it *should* work, because *inside the container* you're running as root. *Publishing* it to the host on ports 25/80 also, because the docker-daemon has root access.

**Pull request merged**